### DISCUSSION TITLE
Text content between tags disappears on DOM update.

### DISCUSSION THREAD

**Discussion by matt-h** [livewire/livewire#3627] [7. Bugs (archived)] [answered]
Text content between around tags disappears on change when you have a new html tag being added to it on livewire updates.

I made up a quick demo that demonstrates this bug: https://github.com/matt-h/livewire-bug

It uses this code:
```
<div>
    <div style="text-align: center;font-size: 3em;">
        @if ($count > 0)
            <button wire:click="decrement" style="font-size: 1em;">-</button>
        @endif
        {{ $count }}
        <button wire:click="increment" style="font-size: 1em;">+</button>
    </div>
</div>
```
```
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class Counter extends Component
{
    public $count = 0;

    public function increment()
    {
        $this->count++;
    }

    public function decrement()
    {
        $this->count--;
    }

    public function render()
    {
        return view('livewire.counter');
    }
}

```

Basically as you increment the counter when it gets to 1 the $count doesn't show when it first adds the decrement button to the DOM. The number is properly being returned the the HTML, but it isn't showing when livewire does its DOM update. If you wrap `{{ $count }}` in a span or any other HTML tag it works fine.

**Comment by ryatkins** [comment, marked as answer]
@matt-h You've come to the write answer by wrapping your content in a span or div

The DOM diffing is losing your content most likely because your button tags look so similar and you're removing the content.

Another option is to disable your button when count > 0 with a disabled tag and you could also style the class this way.
`<button wire:click="decrement" style="font-size: 1em;" @if ($count > 0) disabled @endif >-</button>`

**Reply by matt-h** [reply]
Yeah, I already have a solution for myself. I reported it here as a bug since it seems to be a bug in the diffing engine.

This example isn't at all the actual issue I was having but it was the quickest thing I could build that would re-produce the bug using code from the Livewire docs.

**Reply by matt-h** [reply]
Something else like
```
<div>
    @if ($count > 0)
        <p>Your counter is now more than 0</p>
    @endif
    We've got some text here
    <button wire:click="increment">+</button>
</div>
```

Also produces the same thing where `We've got some text here` will disappear when the `<p>` is added.

**Reply by joshhanley** [reply]
@matt-h it's sort of a bug/ sort of not. The way the diffing works, it's hard for it to handle situations like that without a significant performance overhead, so it's better to wrap conditionals in an element.

It is documented here, have a look at the dom diffing issues section https://laravel-livewire.com/docs/2.x/troubleshooting

Hope this helps!

**Reply by matt-h** [reply]
Okay, no problem then if it is known issue and the solution is to just work around it to avoid the performance impact.
